src/seedu/addressbook/storage/StorageFile.java

```diff
@@ -92,6 +93,13 @@ private static boolean isValidPath(Path filePath) {
      * @throws StorageOperationException if there were errors converting and/or storing data to file.
      */
     public void save(AddressBook addressBook) throws StorageOperationException {
+        
+        // XXX handling of file deleted situation using exceptions.
+        // I think this is a very bad idea. Instead of handling file deleted situation, the
+        // following code will just check if the file exists before saving
+        if (!Files.exists(path)) {
+            throw new StorageOperationException("File does not exist!");
+        }
```

Comment:
Oh I did not realise two things based on what you said.
1. Although the program already handles the situation when file does not exist, it makes sense to inform the user that the file might have gone missing and so they can perhaps investigate if it is due to something like corruption because of hard disk failure.
2. The program handle the situation in  a way that is not clear in the code, perhaps it will also be better to spell out that the program handles a missing file situation.
3. I should have handled the exception so the program does not stop.